std-bits/base/src/main/java/org/enso/base/net/URITransformer.java

```diff
@@ -81,4 +80,27 @@ public static String encodeForQuery(String value) {
   public static String encode(String value) {
     return URLEncoder.encode(value, StandardCharsets.UTF_8).replace("+", "%20");
   }
+
+  public static String encodeQuery(String rawQueryString) {
+    var parts = rawQueryString.split("&");
+    StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < parts.length; i++) {
+      if (i > 0) {
+        sb.append("&");
+      }
+      var indexOfEquals = parts[i].indexOf("=");
+      var keyValue =
+          indexOfEquals == -1
+              ? new String[] {parts[i]}
+              : new String[] {
+                parts[i].substring(0, indexOfEquals), parts[i].substring(indexOfEquals + 1)
+              };
```

Comment:
This can be simplified significantly I think:
```suggestion
      var keyValue = parts.split("=", 2);
```